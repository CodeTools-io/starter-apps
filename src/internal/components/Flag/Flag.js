export default function Flag({ children, name, ...props }) {
  const key = `REACT_APP_FLAG_${name}`;
  const isActivated = process?.env?.[key]?.toLowerCase?.() === 'true';

  if (isActivated) {
    return children;
  }

  return null;
}
