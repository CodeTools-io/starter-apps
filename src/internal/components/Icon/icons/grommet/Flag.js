import React from 'react';
import { Flag } from 'grommet-icons';

export default function FlagIcon({ ...props }) {
  return <Flag {...props} />;
}
