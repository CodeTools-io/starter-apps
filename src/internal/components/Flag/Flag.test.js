import React from 'react';
import Flag from './Flag';

describe('Flag', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <Flag name="EXAMPLE_FLAG">
        <p>Content for a flag that should not be rendered</p>
      </Flag>
    );

    expect(container.child).toMatchSnapshot();
  });
});
