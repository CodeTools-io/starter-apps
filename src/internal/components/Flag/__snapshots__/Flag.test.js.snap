// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag can render 1`] = `undefined`;
